packages/blocks/src/types.ts

```diff
@@ -0,0 +1,977 @@
+	 * dropdown by default. To disable this behavior, set `lock` to `false`.
+	 *
+	 * (default) true
+	 */
+	lock?: boolean | undefined;
+
+	/**
+	 * A block may want to disable the ability to toggle the lock state.
+	 * It can be locked/unlocked by a user from the block “Options”
+	 * dropdown by default. To disable this behavior, set `lock` to `false`.
+	 */
+	typography?: Partial< TypographyProps > | undefined;
+
+	[ key: string ]: any; // Permit untyped/undocumented property access.
+}
+
+//
+// Attributes
+// ----------------------------------------------------------------------------
+
+export namespace AttributeSource {
+	export type Attribute = {
+		source: 'attribute';
+		attribute: string;
+		selector?: string | undefined;
+	} & (
+		| {
+				type: 'boolean';
+				default?: boolean | undefined;
+		  }
+		| {
+				type: 'number';
+				default?: number | undefined;
+		  }
+		| {
+				type: 'string';
+				default?: string | undefined;
+		  }
+	 );
+
+	export interface Children {
+		source: 'children';
+		type: 'array';
+		selector?: string | undefined;
+	}
+
+	export interface HTML {
+		source: 'html';
+		type: 'string';
+		multiline?: 'li' | 'p' | undefined;
+		selector?: string | undefined;
+		default?: string | undefined;
+	}
+
+	export interface Meta {
+		source: 'meta';
+		type: 'string';
+		meta: string;
+		default?: string | undefined;
+	}
+
+	export interface Query< T > {
+		source: 'query';
+		type: 'array';
+		selector: string;
+		query: {
+			[ k in keyof T ]: WPBlockAttribute<
+				T[ k ] extends Array< infer U > ? U : T[ k ]
+			>;
+		};
+		default?: any[] | undefined;
+	}
+
+	export interface Text {
+		source: 'text';
+		type: 'string';
+		selector?: string | undefined;
+		default?: string | undefined;
+	}
+
+	export type None =
+		| ( {
+				source?: never | undefined;
+		  } & (
+				| {
+						type: 'array';
+						default?: any[] | undefined;
+				  }
+				| {
+						type: 'object';
+						default?: object | undefined;
+				  }
+				| {
+						type: 'boolean';
+						default?: boolean | undefined;
+				  }
+				| {
+						type: 'number';
+						default?: number | undefined;
+				  }
+				| {
+						type: 'string';
+						default?: string | undefined;
+				  }
+		   ) )
+		| 'array'
+		| 'object'
+		| 'boolean'
+		| 'number'
+		| 'string';
+}
+
+export type WPBlockAttribute< T > =
+	| AttributeSource.Attribute
+	| AttributeSource.Children
+	| AttributeSource.HTML
+	| AttributeSource.Meta
+	| AttributeSource.Query< T >
+	| AttributeSource.Text
+	| AttributeSource.None
+	| { [ key: string ]: any }; // Permit untyped/undocumented property access.
+
+//
+// Transforms
+// ----------------------------------------------------------------------------
+
+export type BaseTransform = {
+	priority?: number | undefined;
+	__experimentalConvert(
+		blocks: WPBlock | WPBlock[]
+	): WPBlock | WPBlock[] | null;
+	[ key: string ]: any; // Permit untyped/undocumented property access.
+};
```

Comment:
@adamziel Does this look right? 

I just added `BaseTransform` in this commit. Though would `TransformRawSchema` also extend it? It is defined much differently than the others.

Reply:
Looks like BaseTransform is no longer there, I'll resolve this thread